tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc

```diff
@@ -57,8 +57,13 @@ bool GPUBFCAllocator::GetGarbageCollectionValue() {
   const char* enable_gpu_garbage_collection =
       std::getenv("TF_ENABLE_GPU_GARBAGE_COLLECTION");
   if (enable_gpu_garbage_collection == nullptr) {
-    // By default, turn on the memory garbage collection.
-    return true;
+    // By default, turn off the memory garbage collection.
+    //
+    // The mainline TF branch sets this to true. This modification for the
+    // DirectML fork is false so the garbage collector doesn't attempt to
+    // free allocations in the middle of GPU processing, which would result
+    // in a DXGI device removed error and process death.
+    return false;
```

Comment:
Can we add a boolean in the constructor (e.g. `default_garbage_collection_value`) that defaults to `true`, and call it with `false` when calling it from the DML BFC allocator constructor? This is not blocking, but even if we won't merge this upstream it would make it easier to port over to TF2.

Reply:
A fine idea. Added.